src/wallet/wallet-generation.ts

```diff
@@ -0,0 +1,48 @@
+import axios from 'axios'
+
+import {errors} from '../common'
+import {GeneratedAddress} from '../offline/generate-address'
+
+export interface FaucetWallet {
+  accounts: GeneratedAddress
+  amount: number
+  balance: number
+}
+
+const TEST_FAUCET_URL = 'https://faucet.altnet.rippletest.net/accounts'
+const DEV_FAUCET_URL = 'https://faucet.devnet.rippletest.net/accounts'
+
+/**
+ * Generates a random wallet with some amount of XRP (usually 1000 XRP).
+ *
+ * @param onTestnet - If true (default), generates the wallet on the Testnet, otherwise on the Devnet
+ * @returns - A Wallet on the Testnet or Devnet that contains some amount of XRP.
+ */
+function generateFaucetWallet(onTestnet = true): Promise<FaucetWallet | void> {
+  const faucetUrl = getFaucetUrl(onTestnet)
+
+  return axios
+    .post<FaucetWallet>(faucetUrl)
+    .then((response) => {
+      return response.data
+    })
+    .catch((error) => {
+      throw new errors.XRPLFaucetError(error.data)
+    })
+}
```

Comment:
This method isn't going to be very robust, because it takes a few seconds for the transaction to be confirmed on the ledger, so you can't use it immediately. See https://github.com/XRPLF/xrpl-py/blob/master/xrpl/asyncio/wallet/wallet_generation.py#L26

Reply:
Yes, the function returns a `Promise` so my idea was that the user would use `async/await` to get the Wallet information (or use `.then()`).
What do you think?

Reply:
So, do you mean that when you get the response from the POST request, the response includes a balance of 1000 XRP but it can happen that the XRPL account doesn't have 1000 XRP yet?

Reply:
Yeah exactly.

Reply:
Cool thank you, I learnt a lot here.
I made a lot of changes to make sure the account is funded before the function returns the faucet account similar to xrpl-py. Let me know what you think.